xdsl/transforms/memref_to_dsd.py

```diff
@@ -119,7 +123,48 @@ class LowerSubviewOpPass(RewritePattern):
 
     @op_type_rewrite_pattern
     def match_and_rewrite(self, op: memref.SubviewOp, rewriter: PatternRewriter, /):
-        assert isa(op.source.type, MemRefType[Attribute])
+        assert isa(op.source.type, AnyMemRefType)
+        assert isa(op.result.type, AnyMemRefType)
+
+        if len(op.result.type.get_shape()) == 1 and len(op.source.type.get_shape()) > 1:
+            # 1d subview onto a nd memref
+            sizes = op.static_sizes.get_values()
+            counter_sizes = collections.Counter(sizes)
```

Comment:
I don't really understand why the `Counter` is needed here

Reply:
To assert that all but one values are 1, to find the value that isn't 1, and later to find its index. Essentially, to find the active dimension. Plus some asserting that only one dimension is active, as we're creating a 1-d subview on an n-d memref.